### ISSUE TITLE
[Expo

### ISSUE THREAD

**Issue by KMNowak** [flyerhq/react-native-firebase-chat-core#9]
# General

## What bug do you experience? 🐞
Provide a clear and concise description of what the bug is.

## How can it be reproduced? 🤔
A few steps to define where does the bug occur.
Step 1. ...
Step 2. ... etc.

## What behavior is expected? 💡

<br />

# Extras

## Screenshots or videos 📸
If applicable, add screenshots or videos to help explain your problem.

## Code snippets 📝
If applicable, add code samples to help explain your problem.

```ts
// Your snippet here...
```

<br />

# Environment info

Please specify the react, react-native, react-native-firebase-chat-core versions.

react: ...
react-native: ...
react-native-firebase-chat-core: ...

`npx react-native info` output 👇

```sh
```

<br />

# Platform

Device (e.g. Android emulator, iOS simulator, Samsung Galaxy S21): ...

OS version (e.g. iOS 14.5, Android 11): ...

<br />

# Additional context

<!-- Thank you for making us better 🤝--!>

**Comment by cristaloleg** [comment]
Looks like issue is missing some details, can you add?

**Comment by KMNowak** [comment]
Sorry, I accidentally hit enter. Now it's complete :)

**Comment by KMNowak** [comment]
Now it's complete. I'll be grateful if you can give some insights what can be changed to make it work.

**Comment by aaryashahk** [comment]
Having the same issue. Correct me if I am wrong, but I believe it has something to do with FlyerChat using React Native Firebase, which is not usable on the managed workflow utilized by Expo.

**Comment by demchenkoalex** [comment]
Oops completely missed that! We are working on a big update for React Native with new features and Firebase package will be updated too. I will take a look at Expo too.

**Comment by aaryashahk** [comment]
> Oops completely missed that! We are working on a big update for React Native with new features and Firebase package will be updated too. I will take a look at Expo too.

When can we expect this update?

**Comment by demchenkoalex** [comment]
@aaryashahk I have never used Expo before, so I can't suggest anything. PR with an update is already in the UI repo, I just need to find some time to review it, so in the near future.

**Comment by aaryashahk** [comment]
> @aaryashahk I have never used Expo before, so I can't suggest anything. PR with an update is already in the UI repo, I just need to find some time to review it, so in the near future.

I understand. Does this PR with the update that's already in the UI repo have the capability of working with Firebase in Expo's managed workflow?

**Comment by demchenkoalex** [comment]
@aaryashahk UI and Firebase are not related packages. We separated them completely in order for people to use whatever they want, not requiring the other part. From what I know UI has no problems with Expo whatsoever.

**Comment by aaryashahk** [comment]
> @aaryashahk UI and Firebase are not related packages. We separated them completely in order for people to use whatever they want, not requiring the other part. From what I know UI has no problems with Expo whatsoever.

Yes, UI works perfectly fine on Expo. Incorporating Firebase is the problem in Expo since FlyerChat is using react-native-firebase, which is not compatible with Expo's managed workflow. Are you going to add support for FlyerChat's Firebase BaaS to work on Expo's managed workflow?

**Comment by demchenkoalex** [comment]
@aaryashahk you are saying that Expo does not support `react-native-firebase`? That means we will not add this support, because this is the only library we can use for RN

**Comment by gwbischof** [comment]
I think that expo supports `react-native-firebase`

**Comment by demchenkoalex** [comment]
It should work in a bare workflow anyway? The question was about managed workflow I suppose. To do that I will need to rewrite everything using JS SDK.

**Comment by Hirbod** [comment]
It also works now with the managed workflow using custom dev clients and expo config plugins.

**Comment by lalitdhimaney** [comment]
> It also works now with the managed workflow using custom dev clients and expo config plugins.

Hi, @hirbod I am using expo managed workflow and was trying to implement Firebase messaging but getting the same error "Error: You attempted to use a firebase module that's not installed on your Android project by calling firebase.app()." Can you please let me know what specific steps you took while implementing?
Link: https://rnfirebase.io/messaging/notifications 
Expo SDK: 47, React Native: 0.70.8, react-native-firebase/app: 18.1.0, react-native-firebase/messaging: 18.1.0

**Comment by Noothan-am** [comment]
hey @KMNowak, did you get any solution to fix it ?

**Comment by KMNowak** [comment]
> hey @KMNowak, did you get any solution to fix it ?

Unfortunately not, I was applying that to PoCs I was building and gave this specific one up. Anyway, I hope someone will find a solution, so you will get unstuck.

**Comment by cryptodev6** [comment]
Has anyone fixed it? I keep having the same trouble for a long time

**Comment by demchenkoalex** [comment]
please don't use this package @cryptodev6 - not maintained for a very long time. Just keeping it as a placeholder in case I will be able to return to it.
